test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc

```diff
@@ -363,6 +474,128 @@ TEST_P(GrpcJsonTranscoderIntegrationTest, InvalidJson) {
       R"({ "theme"  "Children" })", {}, {}, Status(),
       Http::TestHeaderMapImpl{{":status", "400"}, {"content-type", "text/plain"}},
       "Expected : between key:value pair.\n", false);
+
+  testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
+      Http::TestHeaderMapImpl{{":method", "POST"}, {":path", "/shelf"}, {":authority", "host"}},
+      R"({ "theme" : "Children" }EXTRA)", {}, {}, Status(),
+      Http::TestHeaderMapImpl{{":status", "400"}, {"content-type", "text/plain"}},
+      "Parsing terminated before end of input.\n", false);
+}
+
+std::string createDeepJson(int level, bool valid) {
```

Comment:
can you just create protobuf Struct instead of Json? seems eventually this converted to that.

Reply:
The point is that I'm not confident you're generating a valid JSON here with this code as a reviewer. If you use Struct and build it with Struct semantics (i.e. without concat strings), and convert to JSON at the end, I'm sure that is a valid JSON. As we don't want to add more rapidjson dependency here that is the way to go.